File: backend/Jarvis.Api/Models/LearningLog.cs
namespace Jarvis.Api.Models;

public class LearningLog
{
    public int Id { get; set; }

    public string Topic { get; set; } = string.Empty;

    public string Category { get; set; } = string.Empty;

    public string? Notes { get; set; }

    public int Difficulty { get; set; }

    public DateTime? CompletedAt { get; set; }

    public DateTime CreatedAt { get; set; }
}

File: frontend/angular-app/src/app/models/learning-log.ts
export type LearningCategory =
  | 'CSharp'
  | 'DotNet'
  | 'AspNetCore'
  | 'Sql'
  | 'Angular'
  | 'TypeScript'
  | 'SystemDesign'
  | 'GitHub';

export type LearningTopicStatus = 'NotStarted' | 'InProgress' | 'Completed';

export interface LearningLog {
  id: number;
  topic: string;
  category: LearningCategory;
  notes?: string | null;
  difficulty: number;
  completedAt?: string | null;
  createdAt: string;
}

export interface LearningTodayPlan {
  date: string;
  topic: string;
  category: LearningCategory;
  practicePrompt: string;
  todayLogCount: number;
}

export interface LearningRoadmap {
  categories: LearningRoadmapCategory[];
}

export interface LearningRoadmapCategory {
  name: LearningCategory;
  topics: LearningRoadmapTopic[];
}

export interface LearningRoadmapTopic {
  topic: string;
  status: LearningTopicStatus;
  confidenceLevel: number;
  lastPracticedAt?: string | null;
}
